library/iFixit/Matryoshka/APCu.php

```diff
@@ -0,0 +1,79 @@
+<?php
+
+namespace iFixit\Matryoshka;
+
+use iFixit\Matryoshka;
+
+class APCu extends Backend {
+   public static function isAvailable() {
+      return function_exists('apcu_enabled') && apcu_enabled();
+   }
+
+   public function set($key, $value, $expiration = 0) {
+      return apcu_store($key, $value, $expiration);
+   }
+
+   public function setMultiple(array $values, $expiration = 0) {
+      return apcu_store($values, null, $expiration);
+   }
+
+   public function add($key, $value, $expiration = 0) {
+      return apcu_add($key, $value, $expiration);
+   }
+
+   public function increment($key, $amount = 1, $expiration = 0) {
+      $value = apcu_inc($key, $amount, $success);
+
+      // Call set() if the key doesn't exist.
+      if ($success) {
+         return $value;
+      } else if ($this->set($key, $amount, $expiration) !== false) {
+         return $amount;
+      } else {
+         return false;
+      }
+   }
+
+   public function decrement($key, $amount = 1, $expiration = 0) {
+      return $this->increment($key, -$amount, $expiration);
+   }
+
+   public function get($key) {
+      $value = apcu_fetch($key, $success);
+
+      return $success ? $value : self::MISS;
+   }
+
+   public function getMultiple(array $keys) {
+      if (empty($keys)) {
+         return [[],[]];
+      }
+
+      // Default to an empty array in case no keys were found.
+      $hits = apcu_fetch(array_keys($keys)) ?: [];
+
+      $found = [];
+      $missed = [];
+      foreach ($keys as $key => $id) {
+         $value = array_key_exists($key, $hits) ? $hits[$key] : self::MISS;
+         $found[$key] = $value;
+
+         if ($value === self::MISS) {
+            $missed[$key] = $id;
```

Comment:
It's a little unintuitive but including the misses in `$found` makes [`getAndSetMultiple()`](https://github.com/iFixit/Matryoshka/blob/d8edbd22c8639a827dfe327441033f0362ecac6c/library/iFixit/Matryoshka/Backend.php#L159) easy and efficient because you can merely overwrite the missed values with the new values rather than creating a whole new array. This only works because the keys in `$found` are in the same order as the provided `$keys` array so overwriting the values maintains the same order.

This implementation was heavily influenced by making `getAndSetMultiple()` efficient while still making `getMultiple()` useful on its own. If you don't want the misses from `getMultiple()` then merely `array_filter()` them out.